python/tests/test_writer.py

```diff
@@ -871,6 +872,11 @@ def test_replace_where_overwrite(
     filter_string: str,
 ):
     table_path = tmp_path / sub_path
+
+    # Sometimes the path is already there resulting in failed test
+    if table_path.exists():
```

Comment:
For some reason replacewhere overwrite does not work with a partitioned table, if you remove the partioning then it won't throw this table already exist erorr.

Reply:
Not necessarily. It only happens sometimes, not always. See here the other run on Python 3.10 - 2 passed, 1 failed: https://github.com/delta-io/delta-rs/actions/runs/7356976767/job/20027915530?pr=1996#step:9:261
I thought it was because of existing path because that fixed the issue on my machine but looks like not

Reply:
it could be that there is a race condition somewhere related to partitioning. I'll investigate but for now I'll comment out the paritioning to get the full build

Reply:
Problem found, was an existing code for constraints, fix is on the way

Reply:
Nice find!